_rest_config.php

```diff
@@ -289,6 +289,47 @@ public static function authorization_check($section, $value): void
         }
     }
 
+    // Main function to check scope
+    //  Use cases:
+    //     Only sending $first would be for something like 'openid'
+    //     For using all 3 parameters would be for something like 'user/Organization.write'
+    //       $first = 'user', $second = 'Organization', $third = 'write'
+    //     Note the 3 parameter method will also take into account wildcard scopes.
+    public static function scope_check($first, $second = null, $third = null): void
+    {
+        if (!empty($GLOBALS['oauth_scopes'])) {
+            // Need to ensure has scope
+            if (empty($second)) {
+                // Simply check to see if $first is an allowed scope
+                $scope = $first;
+                if (!in_array($scope, $GLOBALS['oauth_scopes'])) {
+                    http_response_code(401);
+                    exit;
+                }
+            } else {
+                // More complicated 3 degree scope check with wildcard support
+                $scope[] = $first . '/' . $second . '.' . $third;
+                $scope[] = $first . '/*.' . $third;
+                $scope[] = $first . '/' . $second . '.*';
+                $scope[] = $first . '/*.*';
+                $scope[] = '*/' . $second . '.' . $third;
```

Comment:
So I would recommend we don't allow wildcard scopes.  In the OAUTH2 authorization a client can send us a wildcard scope request but we will send back a specific enumerated list of scopes for the resources we are granting them right @sjpadgett?  So by the time a user makes a request against the REST api they won't be sending us any wildcard scopes because their access token that we generate will have only an enumerated list of specific scope grants.

The specification says if we support wildcard access tokens we have to provide both current access as well as grant access to all FUTURE resources / permissions which I think can be a pretty big security whole if we aren't careful.  That's why I recommend we don't handle wildcards. 

Hopefully I'm making sense there.